schema.go

```diff
@@ -349,10 +355,29 @@ type structNode struct {
 	fields []structField
 }
 
-func structNodeOf(t reflect.Type) *structNode {
+type tagSource interface {
+	// Tags extracts parquet-go tags from a struct field. The returned array
+	// indices contains
+	//	0: parquet tag
+	// 	1: parquet-key tag (This is for map keys)
+	// 	2: parquet-value tag (This is for map values)
+	Tags(f *reflect.StructField) [3]string
```

Comment:
I would prefer if we returned a concrete type instead of an array of strings, it would create a more robust API between the parquet package and the extensions.

For example:
```go
type StructTag struct {
 Optional    bool
 Compression compress.Codec
 ... 
}

func ParseStructTag(tag string) (StructTag, error) {
  ...
}
```

Reply:
Sounds good to me. I had something like like this but moved to the protobuf tag provider implementation. Will bring it back here.

Reply:
done c54e4ed

We still pass strings but wrapped in a struct

```go
// StructTag contains struct tag values that are use by parquet-go to configure
// nodes.
type StructTag struct {
	// Value set on parquet struct tag
	Parquet string
	// Value set on parquet-key struct tag
	MapKey string
	// Value set on parquet-value struct tag
	MapValue string
}
```

I am not comfortable leaking tag interpretation outside of `nodeOf` function.  `tagSource` implementors will only need to read `SchemaOf` documentation to know which tag value to return.